File: 0229-majority-element-ii/0229-majority-element-ii.py
class Solution:
    def majorityElement(self, nums):
        countEl = {}
        ans = []
        mini = len(nums) // 3
        
        # Count the occurrences of each element
        for num in nums:
            countEl[num] = countEl.get(num, 0) + 1
        
        # Add elements that appear more than len(nums) // 3 times
        for num, count in countEl.items():
            if count > mini:
                ans.append(num)
        
        return ans

File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        HashMap<Integer,Integer> countEl = new HashMap<>();
        
        List<Integer> ans = new ArrayList<>();
        int mini = (int)Math.floor(nums.length / 3) + 1;
        for(int i = 0; i < nums.length; i++) {
            countEl.put(nums[i], countEl.getOrDefault(nums[i],0) + 1);
            if(countEl.get(nums[i]) == mini) ans.add(nums[i]);
        }
        return ans;
    }
}